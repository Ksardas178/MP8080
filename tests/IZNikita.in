LXI D o100 o000	//Задаем адрес двоичного числа
LXI B o200 o000 //Задаем адрес результата
//MVI D 10 //Инициализируем счетчик
//MVI E 0 //Переменная под максимальное число

LDAX D	//Загружаем число в аккумулятор

>LABEL
	
	
	SUB E	//A <- A - E
	JM 14 0//if (A < 0) goto ENDIF
		//else store new max element
		LDAX B	//A <- arr[i]
		MOV E A	//E <- A*
	>ENDIF

	INX B	//i++	
	DCR D 	//D--
JNZ LABEL 	//if (D != 0) goto LABEL

/*Store max value to addr*/	
MVI A E
STA o100 o000

HLT