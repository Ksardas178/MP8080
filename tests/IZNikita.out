stored label LABEL on line 7
stored label ENDIF on line 13

[32;1mInput is correct
[0m
[30;1mCode analysis results:
[0m______________________________
00000000: 00010001
00000001: 01000000
00000010: 00000000
00000011: 00000001
00000100: 10000000
00000101: 00000000
00000110: 00011010
00000111: 10010011
00001000: 11111010
00001001: 00001110
00001010: 00000000
00001011: 00001010
00001100: 01011111
00001101: 00000011
00001110: 00010101
00001111: 11000010
00010000: 00000111
00010001: 00000000
00010010: 00111110
00010011: 00000011
00010100: 00110010
00010101: 01000000
00010110: 00000000
00010111: 01110110
